extensions/guacamole-auth-saml/src/main/java/org/apache/guacamole/auth/saml/AuthenticationProviderService.java

```diff
@@ -0,0 +1,202 @@
+                catch (IOException e) {
+                    logger.warn("Error during I/O while parsing SAML response: {}", e.getMessage());
+                    logger.debug("Received IOException when trying to parse SAML response.", e);
+                    throw new GuacamoleInvalidCredentialsException("Error during SAML login.",
+                                CredentialsInfo.USERNAME_PASSWORD);
+                }
+                catch (ParserConfigurationException e) {
+                    logger.warn("Error configuring XML parser: {}", e.getMessage());
+                    logger.debug("Received ParserConfigurationException when trying to parse SAML response.", e);
+                    throw new GuacamoleInvalidCredentialsException("Error during SAML login.",
+                                CredentialsInfo.USERNAME_PASSWORD);
+                }
+                catch (SAXException e) {
+                    logger.warn("Bad XML when parsing SAML response: {}", e.getMessage());
+                    logger.debug("Received SAXException while parsing SAML response.", e);
+                    throw new GuacamoleInvalidCredentialsException("Error during SAML login.",
+                                CredentialsInfo.USERNAME_PASSWORD);
+                }
+                catch (SettingsException e) {
+                    logger.warn("Error with SAML settings while parsing response: {}", e.getMessage());
+                    logger.debug("Received SettingsException while parsing SAML response.", e);
+                    throw new GuacamoleInvalidCredentialsException("Error during SAML login.",
+                                CredentialsInfo.USERNAME_PASSWORD);
+                }
+                catch (ValidationError e) {
+                    logger.warn("Error validating SAML response: {}", e.getMessage());
+                    logger.debug("Received ValidationError while parsing SAML response.", e);
+                    throw new GuacamoleInvalidCredentialsException("Error during SAML login.",
+                                CredentialsInfo.USERNAME_PASSWORD);
+                }
+                catch (XPathExpressionException e) {
+                    logger.warn("Problem with XML parsing response: {}", e.getMessage());
+                    logger.debug("Received XPathExpressionException while processing SAML response.", e);
+                    throw new GuacamoleInvalidCredentialsException("Error during SAML login.",
+                                CredentialsInfo.USERNAME_PASSWORD);
+                }
+                catch (Exception e) {
+                    logger.warn("Exception while getting name from SAML response: {}", e.getMessage());
+                    logger.debug("Received Exception while retrieving name from SAML response.", e);
+                    throw new GuacamoleInvalidCredentialsException("Error during SAML login.",
+                                CredentialsInfo.USERNAME_PASSWORD);
+                }
```

Comment:
Wow, this is nasty.  Unfortunately the SAML client doesn't currently capture exceptions it generates, so we have to do it, here.

Reply:
Will just keep this as-is for now - better at this point to capture the exceptions with specific error messages than to lump them all together.